libr/anal/p/anal_sh.c

```diff
@@ -196,10 +196,13 @@ static ut64 disarm_12bit_offset (RAnalOp *op, unsigned int insoff) {
 
 /* for bt,bf sign-extended offsets : return PC+4+ (exts.b offset)<<1 */
 static ut64 disarm_8bit_offset (ut64 pc, ut32 offs) {
+        /* pc (really, op->addr) is 64 bits, so we need to sign-extend
+         * to 64 bits instead of the 32 the actual CPU does */
+        ut64 off = offs;
 	/* sign extend if higher bit is 1 (0x08) */
-	if ((offs & 0x80) == 0x80)
-		offs |= ~0xFF;
-	return (offs<<1) + pc + 4;
+	if ((off & 0x80) == 0x80)
```

Comment:
The original code didn't have them, but I went ahead and added braces to this and disarm_12bit_offset :)